src/xml.rs

```diff
@@ -27,221 +33,429 @@ pub struct Document {
+            Self::Blue => &Self::RGB_BLUE,
+            Self::Violet => &Self::RGB_VIOLET,
+            Self::Custom(rgb_color) => rgb_color,
+        }
+    }
+
+    #[must_use]
+    fn from_hex<S: AsRef<str>>(value: S) -> Option<RGB8> {
+        let hexstr = value
+            .as_ref()
+            .strip_prefix("#")
+            .or_else(|| value.as_ref().strip_prefix("0x"))
+            .unwrap_or(value.as_ref());
+        match hexstr.len() {
+            3 => {
+                let mut r = u8::from_str_radix(hexstr.get(0..1)?, 16).ok()?;
+                r |= r << 4;
+                let mut g = u8::from_str_radix(hexstr.get(1..2)?, 16).ok()?;
+                g |= g << 4;
+                let mut b = u8::from_str_radix(hexstr.get(2..3)?, 16).ok()?;
+                b |= b << 4;
+                Some(RGB8 { r, g, b })
+            }
+            6 => {
+                let r = u8::from_str_radix(hexstr.get(0..=1)?, 16).ok()?;
+                let g = u8::from_str_radix(hexstr.get(2..=3)?, 16).ok()?;
+                let b = u8::from_str_radix(hexstr.get(4..=5)?, 16).ok()?;
+                Some(RGB8 { r, g, b })
+            }
+            _ => None,
+        }
+    }
+}
+
+impl From<RGB8> for Color {
+    fn from(rgb_color: RGB8) -> Self {
+        match rgb_color {
+            Self::RGB_ROSE => Self::Rose,
+            Self::RGB_RED => Self::Red,
+            Self::RGB_ORANGE => Self::Orange,
+            Self::RGB_LEMON => Self::Lemon,
+            Self::RGB_GREEN => Self::Green,
+            Self::RGB_TURQUOISE => Self::Turquoise,
+            Self::RGB_BLUE => Self::Blue,
+            Self::RGB_VIOLET => Self::Violet,
+            rgb_color => Self::Custom(rgb_color),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for Color {
+    type Error = &'a str;
+
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        match value {
+            "Rose" => Ok(Self::Rose),
+            "Red" => Ok(Self::Red),
+            "Orange" => Ok(Self::Orange),
+            "Lemon" => Ok(Self::Lemon),
+            "Green" => Ok(Self::Green),
+            "Turquoise" => Ok(Self::Turquoise),
+            "Blue" => Ok(Self::Blue),
+            "Violet" => Ok(Self::Violet),
+            color_str => Color::from_hex(color_str)
+                .map(Color::Custom)
+                .ok_or(value.as_ref()),
+        }
+    }
+}
+
+impl fmt::Display for Color {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Rose => write!(f, "Rose"),
+            Self::Red => write!(f, "Red"),
+            Self::Orange => write!(f, "Orange"),
+            Self::Lemon => write!(f, "Lemon"),
+            Self::Green => write!(f, "Green"),
+            Self::Turquoise => write!(f, "Turquoise"),
+            Self::Blue => write!(f, "Blue"),
+            Self::Violet => write!(f, "Violet"),
+            Self::Custom(rgb) => write!(f, "#{:02X}{:02X}{:02X}", rgb.r, rgb.g, rgb.b),
```

Comment:
Alternative: delegate to the inner type if you don't need a stable representation 
https://docs.rs/rgb/latest/rgb/type.RGB8.html#impl-Display-for-Rgb%3CT%3E
```suggestion
            Self::Custom(rgb) => write!(f, "{}", rgb),
```